File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    notification_type: str
    title: str
    message: str
    related_id: uuid.UUID | None
    is_read: bool
    created_at: datetime


class NotificationUpdate(BaseModel):
    is_read: bool


class NotificationUnreadCount(BaseModel):
    unread_count: int

File: frontend/src/types/notification.ts
export type NotificationType = 'study_reminder' | 'deadline_reminder' | 'quiz_reminder' | 'system'

export interface Notification {
  id: string
  notification_type: NotificationType
  title: string
  message: string
  related_id: string | null
  is_read: boolean
  created_at: string
}

export interface UnreadCount {
  unread_count: number
}